scipy/optimize/tests/test_milp.py

```diff
@@ -271,3 +271,31 @@ def test_infeasible_prob_16609():
     res = milp(c, integrality=integrality, bounds=bounds,
                constraints=constraints)
     np.testing.assert_equal(res.status, 2)
+
+
+def test_milp_timeout_16545():
+    # Ensure solution is not thrown away if MILP solver times out
+    # -- see gh-16545
+    np.random.seed(16545)
+    A = np.zeros((100, 100))
+    for c in range(100):
+        for v in range(100):
+            A[c, v] = np.random.randint(0, 5)
+    b_lb = [-1 * np.inf for c in range(100)]
+    b_ub = [25 for c in range(100)]
+    constraints = LinearConstraint(A, b_lb, b_ub)
+    variable_lb = [0 for v in range(100)]
+    variable_ub = [1 for v in range(100)]
+    variable_bounds = Bounds(variable_lb, variable_ub)
+    integrality = [1 for v in range(100)]
+    c_vector = 100 * [-1]
+    res = milp(
+        c_vector,
+        integrality=integrality,
+        bounds=variable_bounds,
+        constraints=constraints,
+        options={"time_limit": .01}
+    )
+    msg = "Time limit reached. (HiGHS Status 13:"
+    assert res.message.startswith(msg)
+    assert res["x"] is not None
```

Comment:
Could you show feasibility?
And can we parametrize to show that we get a solution when the iteration limit is reached?

Reply:
The other time limit test does assert that `res['x'] is None` when it has not found a feasible solution.  I can add in a check to ensure feasibility, yes

Reply:
Looking into iteration limit now

Reply:
I went ahead and added `mip_max_nodes` as a first-class option (hopefully didn't jump the gun on this one) and added the suggest parameterization to the test

Reply:
Is that the thing that corresponds with `HighsModelStatusREACHED_ITERATION_LIMIT`?

Reply:
> Is that the thing that corresponds with `HighsModelStatusREACHED_ITERATION_LIMIT`? That's the part of the changes I wanted to test.

Yes, `mip_max_nodes` being reached will stop the MIP solver and return `HighsModelStatusREACHED_ITERATION_LIMIT`

Reply:
Is `mip_max_nodes` what HiGHS calls the option? If so, I guess it's fine. I just think it will be surprising to a user that if they set `mip_max_nodes` too low, they get an issue about reaching an "iteration" limit.

Reply:
`mip_max_nodes` is the HiGHS option, but we can alias it to something else if it will be clearer.  From a cursory look around, "iteration limit" almost always applies to the simplex solvers, not the MIP solver (e.g., [Gurobi](https://www.gurobi.com/documentation/9.5/refman/parameters.html) has both an `IterationLimit` and `NodeLimit`), so I think it might be confusing to use "iteration limit"

Reply:
OK, I made it `node_limit`.